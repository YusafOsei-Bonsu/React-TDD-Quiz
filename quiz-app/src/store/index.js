import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

// Setting up storage to make sure the store persist after refresh
const persistConfig = {
    key: 'root',
    storage,
};
// Initialise store
const initialState = {
    users: [],
    catDropDown: [],
    quizData: []
}
// Initialize reducer. The reducer is the only way to communicate with state.
// It takes an action creates new state object, then updates the new state object. 
const reducer = (state = initialState, action) => {
    console.log(action.payload)
    switch (action.type) {
        case 'catDropDown':
            // console.log( action.payload)
            return Object.assign({}, state, { catDropDown: action.payload })
        case 'questionsAnswers':
            // console.log( action.payload)
            return Object.assign({}, state, { quizData: action.payload })
        case 'addUser':
            return Object.assign({}, state, { users: action.payload })
        default:
            return state
    }
}
const persistedReducer = persistReducer(persistConfig, reducer)
// Create and export store
export const store = createStore(persistedReducer, applyMiddleware(thunk));
// Export persistent store
export let persistor = persistStore(store);