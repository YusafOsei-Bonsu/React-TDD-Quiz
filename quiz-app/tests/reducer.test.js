import {reducer, initialState} from '../src/store/index';

describe('Reducer', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, {})).toMatchSnapshot()
    })
    it('should handle addUser', () => {
      expect(
        reducer(initialState,
        {
          type: 'addUser'
        })
      ).toMatchSnapshot()
    })
    it('should handle addingCatData', () => {
      expect(
        reducer(initialState,
        {
          type: 'catDropDown'
        })
      ).toMatchSnapshot()
    })
    it('should handle addingQuizData', () => {
      expect(
        reducer(initialState,
        {
          type: 'quizData'
        })
      ).toMatchSnapshot()
    })
    it('should handle error', () => {
        const failAction = {
          type: 'handle-error',
          hasErrors: false 
        };
        expect(reducer({}, failAction)).toEqual({hasErrors :true});
      });
    it('should handle increment', () => {
        const failAction = {
          type: 'increment-error'
        };
        expect(reducer({}, failAction)).toEqual({});
      });
  })