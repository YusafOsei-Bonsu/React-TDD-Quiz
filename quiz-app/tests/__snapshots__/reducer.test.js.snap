// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer should handle TODO_REQUEST 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`Reducer should handle addUser 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": undefined,
}
`;

exports[`Reducer should handle addUser 2`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": undefined,
}
`;

exports[`Reducer should handle adding 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": undefined,
}
`;

exports[`Reducer should handle addingCatData 1`] = `
Object {
  "catDropDown": undefined,
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`Reducer should handle addingData 1`] = `
Object {
  "catDropDown": undefined,
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`Reducer should handle addingData 2`] = `
Object {
  "catDropDown": undefined,
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`Reducer should handle addingQuizData 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`Reducer should handle category 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": true,
  "quizData": Array [],
  "score": true,
  "users": Array [],
}
`;

exports[`Reducer should handle category data deli 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": true,
  "quizData": Array [],
  "score": true,
  "users": Array [],
}
`;

exports[`Reducer should handle category data delivery 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": true,
  "quizData": Array [],
  "score": true,
  "users": Array [],
}
`;

exports[`Reducer should handle error 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": undefined,
}
`;

exports[`Reducer should return the initial state 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`todos reducer should handle TODO_REQUEST 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;

exports[`todos reducer should return the initial state 1`] = `
Object {
  "catDropDown": Array [],
  "hasErrors": false,
  "quizData": Array [],
  "score": 0,
  "users": Array [],
}
`;
